_posts/2021-03-30-influxdb-rails.html

```diff
@@ -0,0 +1,199 @@
+      </li>
+    </ul>
+  </p>
+</div>
+
+<h2>Application Performance Monitoring (APM)</h2>
+<p>
+  I'm not sure we have to discuss this at all anymore, but here is why I think I need application performance
+  monitoring for my Ruby on Rail apps: I'm an expert in using the software I hack. Hence I always travel the happy
+  path, cleverly and unconsciously avoid the pitfalls, use it with forethought of the underlying architecture.
+  Because of that, I usually <b>think</b> my app is fast: <a href="https://blog.teamtreehouse.com/perceived-performance">perceived performance</a>.
+</p>
+
+<p>
+  <blockquote>
+    It's not what you look at that matters, it's what you see.
+    – Henry David Thoreau
+  </blockquote>
+</p>
+
+<p>
+  That is why I need someone to correct that BIAS for me, in black and white.
+</p>
+
+<h2>Free Software APM</h2>
+
+<p>
+  The good folks at <a href="https://contributors.rubyonrails.org/">Rails</a> bring the instrumentation framework.
+  <a href="https://www.influxdata.com/about/">InfluxData</a> deliver a time series database.
+  <a href="https://grafana.com/about/team/">Grafana Labs</a> make a dashboard builder. All we need to do, as so often,
+  is to plug Free Software together: SUCCESS!
+</p>
+
+<h3>InfluxDB + Grafana == 🧨</h3>
+
+<p>
+  I assume your Rails development environment is already running on 127.0.0.0:300. Getting InfluxDB and Grafana is a matter
+  of pulling containers these days. You can use this simple <code>docker-compose</code> configuration for running them locally
+  too.
+</p>
+
+{% highlight yaml %}
+# docker-compose.yml
+version: "3.7"
+services:
+  influx:
+    image: influxdb:1.7
+    environment:
+      - INFLUXDB_DB=rails
+      - INFLUXDB_USER=root
+      - INFLUXDB_USER_PASSWORD=root
+      - INFLUXDB_READ_USER=grafana
+      - INFLUXDB_READ_USER_PASSWORD=grafana
+    volumes:
+      - influxdata:/var/lib/influxdb
+    ports:
+      - 8086:8086
+  grafana:
+    image: grafana/grafana:7.4.5
+    ports:
+      - 4000:3000
+    depends_on:
+      - influx
+    volumes:
+      - grafanadata:/var/lib/grafana
+volumes:
+  influxdata:
+  grafanadata:
+{% endhighlight %}
+
+<p>
+  A courageous <code>docker-compose up</code> will boot things and you can access Grafana at <a href="http://127.0.0.1:4000">http://127.0.0.1:4000</a> (user: admin / password: admin).
+  To read data from the InfluxDB container in Grafana, leave the <a href="http://127.0.0.1:4000/datasources">/datasources</a>
+  InfluxDB defaults alone and configure:
+</p>
+
+{% highlight shell %}
+URL: http://influx:8086
+Database: rails
+User: grafana
+Password: grafana
+{% endhighlight %}
+
+<h3>influxdb-rails: 🪢 things together</h3>
+<p>
+  The <code>influxdb-rails</code> RubyGem is the missing glue code for making your app report metrics into the InfluxDB.
+  Plug it into your Rails app by adding it to your bundle.
+</p>
+
+{% highlight shell %}
+echo 'gem "influxdb-rails"' >> Gemfile
```

Comment:
This line and `bundle install` can be replaced by this:

````suggestion
bundler add influxdb-rails
```

Reply:
Cool, didn't know that one!